### PR TITLE
fix(migrations): do not migrate for libs & migrate for server #12508

### LINKED ISSUE #12508: Wrong angular.json migration for projectType="library"
## Description  
The `angular.json` migration with 15.0.4 adds `stylesPreprocessorOptions` to projects of type `library`.

 * igniteui-angular version: 15.0.4
 * browser: N/A

## Steps to reproduce  

1. Execute migrations on a library project depending on `igniteui-angular`

## Result  
The `stylesPreprocessorOptions` is added.

## Expected result  
It shouldn't.

## Attachments  
image

### PR DESCRIPTION
**Pull request by Lipata**
Closes #12508

### Additional information (check all that apply):
 - [x] Bug fix
 - [ ] New functionality
 - [ ] Documentation
 - [ ] Demos
 - [ ] CI/CD

### Checklist:
 - [x] All relevant tags have been applied to this PR
 - [x] This PR includes unit tests covering all the new code ([test guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Test-implementation-guidelines-for-Ignite-UI-for-Angular))
 - [ ] This PR includes API docs for newly added methods/properties ([api docs guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Documentation-Guidelines))
 - [ ] This PR includes `feature/README.MD` updates for the feature docs
 - [ ] This PR includes general feature table updates in the root `README.MD`
 - [ ] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [ ] This PR contains breaking changes
 - [ ] This PR includes `ng update` migrations for the breaking changes ([migrations guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Update-Migrations))
 - [ ] This PR includes behavioral changes and the feature specification has been updated with them

### CODE REVIEW

File: projects/igniteui-angular/migrations/update-15_0_4/index.spec.ts [hunk 1]
```diff
@@ -24,18 +24,43 @@ describe(`Update to ${version}`, () => {
         },
         version: 1
     };
+    const configJsonLib = {
+        defaultProject: 'testProj',
+        projects: {
+            testProj: {
+                root: '/',
+                sourceRoot: '/testSrc',
+                prefix: 'lib',
```
Comment (kdinev) [inline projects/igniteui-angular/migrations/update-15_0_4/index.spec.ts:33]: The prefix of a project of `projectType: 'library'` could be anything. The prefix of igniteui-angular is `igx`. The project type you're testing with here is the default one, which is `application`.

**Review by kdinev — CHANGES_REQUESTED** [review]

File: projects/igniteui-angular/migrations/update-15_0_4/index.spec.ts (hunk 1 above)
Comment (Lipata) [reply projects/igniteui-angular/migrations/update-15_0_4/index.spec.ts:33]: Yes, in the schematics we had an enumeration that is checking against projectType, not the prefix, so I'm testing aginst it.

File: projects/igniteui-angular/schematics/utils/dependency-handler.ts [hunk 2]
```diff
@@ -158,8 +158,10 @@ const addHammerToConfig =
     };
 
 export const includeStylePreprocessorOptions = async (workspaceHost: workspaces.WorkspaceHost, workspace: workspaces.WorkspaceDefinition, context: SchematicContext, tree: Tree): Promise<void> => {
-    await Promise.all(Array.from(workspace.projects.values()).map(async (project) => {
+    await Promise.all(Array.from(workspace.projects.values()).map(async (project: workspaces.ProjectDefinition) => {
+        if (project.extensions['projectType'] === ProjectType.Library) return;
         await addStylePreprocessorOptions(project, tree, "build", context);
+        await addStylePreprocessorOptions(project, tree, "serve", context);
```
Comment (kdinev) [inline projects/igniteui-angular/schematics/utils/dependency-handler.ts:164]: This does not cover `server` configuration. It covers `serve`, which doesn't need the style preprocessor options.

**Review by kdinev — CHANGES_REQUESTED** [review]

**Review by kdinev — APPROVED** [review]
Looks good, but make sure you test the server configuration migration in a project with SSR.

**Comment by Lipata** [comment]
@kdinev after I've added SSR to a ng project with `ng add @nguniversal/express-engine`, then the `server` config in `angular.json` was migrated properly. The project then can be built and is running properly with `npm run dev:ssr`.

**Pull request merged**